packages/Discretization/test/PointCloudProblemGenerator/ExodusAsciiGenerator_impl.hpp

```diff
@@ -0,0 +1,321 @@
+// mapping.
+template <class SourceDevice, class TargetDevice>
+template <class Device>
+void ExodusAsciiGenerator<SourceDevice, TargetDevice>::
+    partitionGhostedConnectivity(
+        const int dim, const std::string &coord_file,
+        const std::string &connectivity_file,
+        Kokkos::View<Coordinate **, Kokkos::LayoutLeft, Device> &device_coords,
+        Kokkos::View<GlobalOrdinal *, Kokkos::LayoutLeft, Device> &device_gids )
+{
+    // Get the node data.
+    Kokkos::View<Coordinate **, Kokkos::LayoutLeft, Kokkos::Serial> host_coords;
+    Kokkos::View<GlobalOrdinal *, Kokkos::LayoutLeft, Kokkos::Serial> host_gids;
+    getNodeDataFromFile( coord_file, host_coords, host_gids );
+
+    // Partition the source coordinates in the dimension. Figure out the min
+    // and max coordinates.
+    Coordinate dim_max, dim_min;
+    dimensionMinAndMax( dim, host_coords, dim_min, dim_max );
+
+    // Load the connectivity file.
+    std::ifstream conn_file;
+    conn_file.open( connectivity_file );
+
+    // Partition based on the dimension coordinate of the first node in each
+    // cell. All nodes belonging to that cell will be sent to that rank.
+    Teuchos::Array<GlobalOrdinal> export_gids( 0 );
+    Teuchos::Array<int> export_ranks( 0 );
+    Teuchos::Array<Coordinate> export_coords( 0 );
+    std::set<std::pair<int, GlobalOrdinal>> unique_exports;
+
+    // Read connectivity data on rank 0.
+    if ( 0 == _comm->getRank() )
+    {
+        int num_cells = 0;
+        conn_file >> num_cells;
+        GlobalOrdinal cell_gid;
+        GlobalOrdinal node_gid;
+        int cell_num_node;
+        std::string cell_topo;
+        double x_frac = 0.0;
+        int send_rank;
+        for ( int c = 0; c < num_cells; ++c )
+        {
+            // Get the cell data.
+            conn_file >> cell_gid;
+            conn_file >> cell_topo;
+            conn_file >> cell_num_node;
+
+            // Get the first cell node.
+            conn_file >> node_gid;
+
+            // Partition.
+            x_frac = ( host_coords( node_gid - 1, dim ) - dim_min ) /
+                     ( dim_max - dim_min );
+            send_rank = ( 1.0 == x_frac )
+                            ? _comm->getSize() - 1
+                            : std::floor( x_frac * _comm->getSize() );
+
+            // Only add this node/rank combo if we haven't already.
+            if ( !unique_exports.count(
+                     std::make_pair( send_rank, node_gid ) ) )
+            {
+                export_gids.push_back( node_gid );
+                export_ranks.push_back( send_rank );
+                export_coords.push_back( host_coords( node_gid - 1, 0 ) );
+                export_coords.push_back( host_coords( node_gid - 1, 1 ) );
+                export_coords.push_back( host_coords( node_gid - 1, 2 ) );
+                unique_exports.insert( std::make_pair( send_rank, node_gid ) );
+            }
```

Comment:
Rather than calling `std::set::count()`, use `::insert`:
```C++
bool inserted = false;
std::tie( std::ignore, inserted ) = unique_exports.insert( ... );
if ( inserted )
{
   ...
}
```